### ISSUE TITLE
Support git parameters

### ISSUE THREAD

**Issue by brandstaetter** [mattmahn/gitall.rs#15]
In
```
> gitall --version
gitall 0.6.0
```

```
> git pull --rebase origin master
From git-server:repository
 * branch            master     -> FETCH_HEAD
Already up to date.
```

but

```
> gitall pull --rebase origin master
error: Found argument '--rebase' which wasn't expected, or isn't valid in this context

USAGE:
    gitall.exe [FLAGS] [OPTIONS] <COMMAND>...

For more information try --help
```

please improve the parser to allow flags and options in the COMMAND as well

**Comment by mattmahn** [comment]
I'm guessing you already know this, but just in case: this can be worked around by putting ` -- ` between gitall options and whatever `<COMMAND>` you're running.
Another option for reducing how much you need to type is to create an `alias gitall='gitall --'` so that you typically don't have to type out the `--` yourself, but you still have access to gitall options by not using the alias: `\gitall -D ~/some/codez -- pull --rebase origin master`.

-----

This would require the CLI argument parser to ignore any options that are not known to gitall.  However, because gitall supports running any executable, this quickly gets into dangerous territory about surprising overlaps with other executables' options.  It also brings up questions about adding new options to gitall.

I admit the current UX isn't perfect, but I'm leaning towards keeping the current behavior because it's less surprising

**Comment by brandstaetter** [comment]
Nope, wasn't aware of the workaround, please add that to the documentation / Help text! I am fine with that solution, thanks for providing this helpful tool!

I'll probably go for `alias gitalls=gitall --` so that I don't have to remember the escaping to circumvent the alias (TIL!)
